### PR TITLE
feat(config): replace config from branches in baseBranches

### PR DESCRIPTION
**Pull request by exoszajzbuk**
## Changes

Add the configuration option `useBaseBranchConfig=replace` 

## Context

The ability to completely replace local repo config files in base branches has been discussed for quite some time, mainly in the following issue:

- #2496

and has also discussions around the topic in:

- https://github.com/renovatebot/renovate/discussions/22264
- https://github.com/renovatebot/renovate/discussions/19916
- https://github.com/renovatebot/renovate/discussions/16108

but the [resulting PR](https://github.com/renovatebot/renovate/pull/12514) only implemented the `merge` strategy for config files on base branches.

This PR intends to add the "missing" `replace` strategy in order to fully replace configuration on base branches independently of the configuration on the repo' default branch.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/config/types.ts [hunk 1]
```diff
@@ -193,7 +193,7 @@ export interface RegExManager extends RegexManagerTemplates {
   autoReplaceStringTemplate?: string;
 }
 
-export type UseBaseBranchConfigType = 'merge' | 'none';
+export type UseBaseBranchConfigType = 'merge' | 'replace' | 'none';
```
Comment (viceice) [inline lib/config/types.ts:196]: same here

File: lib/workers/repository/process/index.ts [hunk 2]
```diff
@@ -71,6 +71,35 @@ async function getBaseBranchConfig(
...
+  else if (
+    config.useBaseBranchConfig === 'replace' &&
+    baseBranch !== config.defaultBranch
+  ) {
+    logger.debug(
+      { baseBranch },
+      `Replacing config from base branch because useBaseBranchConfig=replace`
+    );
+
+    const branchConfigFileName = 'TBD'; // TODO: re-detect config file name on `baseBranch`
```
Comment (viceice) [inline lib/workers/repository/process/index.ts:83]: i would prefer to enforce same file name, as it makes everything simpler. anything else can be done in separate follow up PR.

**Review by viceice — CHANGES_REQUESTED** [review]

File: lib/workers/repository/process/index.ts (hunk 2 above)
Comment (exoszajzbuk) [reply lib/workers/repository/process/index.ts:83]: I understand that it'd be the most simple solution to rely on the already detected filename from the default branch, but I've gathered my thoughts about this limitation on this discussion comment: https://github.com/renovatebot/renovate/discussions/22264#discussioncomment-6111232

if you guys have the time, please read through the rationale behind this "replace" logic running totally independently from the default branch and its configuration

tl;dr - enforcing the same config file name might be an easy solution, but defeats half the purpose of the whole change, meaning to be able to fully test changes in `renovate.json` (and other config files) on side branches, independently of the "stable" configuration running (most probably, but not necessarily) on the default branch

Comment (viceice) [reply lib/workers/repository/process/index.ts:83]: let's do the detection inside a separate PR, so this can be merged faster.

**Review by viceice — COMMENTED** [review]
see comment

**Comment by rarkins** [comment]
@exoszajzbuk do you have time to resume this PR?

**Comment by rarkins** [comment]
Closing due to inactivity
